### PR TITLE
feat: add support for node v20

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all. I also updated our linter and browser tests to use v16 instead of now-deprecated v14.

### Context of Change

node v20 was just released

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Test Plan

CI passes with node 20.

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x]
+        node-version: [14.x, 16.x, 18.x, 20.x]
```
Comment (sublimator) [inline .github/workflows/nodejs.yml:60]: What are the plans re: node 14 which is EOL now ? 
Depending upon those plans, could potentially remove the need for another lib: 
https://github.com/XRPLF/xrpl.js/pull/2273#discussion_r1177267048

Thanks!

Comment (mvadari) [reply .github/workflows/nodejs.yml:60]: You'll have to ask some of the other maintainers about that - it's not my responsibility

**Review by justinr1234 — APPROVED** [review]

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (ckniffen) [reply .github/workflows/nodejs.yml:60]: We should drop in 3.0.

**Review by ckniffen — APPROVED** [review]

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (sublimator) [reply .github/workflows/nodejs.yml:60]: > You'll have to ask some of the other maintainers 

I see, you just had that air of authority :)

**Pull request merged**